projects/RPi/devices/RPi4/bootloader/canupdate.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/sh
+
+UPGRADE_IMG=$2
+
+RPI_REVISION=$(hexdump -ve '1/1 "%.2x"' </proc/device-tree/system/linux,revision)
+RPI_CPU=${RPI_REVISION:4:1}
+
+if echo $UPGRADE_IMG | grep -qE "^RPi.\.*"; then
```

Comment:
```suggestion
if echo $UPGRADE_IMG | grep -qE "^RPi[0-9]?+\..*"; then
```
would correctly match `RPi.arm`, `RPi2.arm`, `RPi4.arm` etc. - the original would fail to match `RPi.arm`and treat it as a non-RPi image.

However you're testing that the upgrade image is a valid RPi image, when this script will only exist in the RPi4 image, so `UPGRADE_IMG` should always be `RPi4.arm` and can't be anything else - would it not make more sense to check if the booted installation ($1) is an RPi system?

Reply:
I believe I follow, and made the suggested changes. Will do run testing tomorrow if not beaten to it.